apps/lockfile-explorer-web/src/containers/LockfileEntryDetailsView/index.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styles from './styles.scss';
+import appStyles from '../../appstyles.scss';
+import { IDependencyType, LockfileDependency } from '../../parsing/LockfileDependency';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { pushToStack, selectCurrentEntry } from '../../store/slices/entrySlice';
+import { ReactNull } from '../../types/ReactNull';
+import { LockfileEntry } from '../../parsing/LockfileEntry';
+
+enum InfluencerTypes {
+  DETERMINANT,
+  TRANSITIVE_REFERRER
+}
+
+interface IInfluencerType {
+  entry: LockfileEntry;
+  type: InfluencerTypes;
+}
+
+export const LockfileEntryDetailsView = (): JSX.Element | ReactNull => {
+  const selectedEntry = useAppSelector(selectCurrentEntry);
+  const dispatch = useAppDispatch();
+
+  const [inspectDep, setInspectDep] = useState<LockfileDependency | null>(null);
+  const [influencers, setInfluencers] = useState<IInfluencerType[]>([]);
+
+  useEffect(() => {
+    if (selectedEntry) {
+      setInspectDep(null);
+    }
+  }, [selectedEntry]);
+
+  const selectResolvedEntry = useCallback(
+    (dependency) => () => {
+      if (inspectDep && inspectDep.name === dependency.name) {
+        if (dependency.resolvedEntry) {
+          dispatch(pushToStack(dependency.resolvedEntry));
+        } else {
+          console.error('No resolved entry for dependency: ', dependency);
```

Comment:
We should think about how to tell the end user about this problem.  For example, maybe the UI would display a panel that says `There were problems loading your lockfile. View the browser's console log for details.`.  Ideally the UI could display these errors in its own panel or dialog box.

We don't need to fix it in this PR, though, just something to think about.

Reply:
I was thinking that this was more for development purposes, to find an error in our logic - for a valid lockfile wouldn't this never happen if the logic for parsing is implemented correctly?

Reply:
Think about it from the perspective of a total stranger who just started working with the project:

Let's say they press F12, and the Chrome console shows  `Error: No resolved entry for dependency` in red text.  Is this a bug?  Or maybe my lockfile is corrupt? Or maybe a new release of PNPM has changed the file format?  Or is this an "error" just part of a huge wall of red text that everybody can safely ignore? 🙂

We don't need to do anything fancy or involved, but we should provide some kind of context for this message.  Some possible approaches:

- If it really is not expected to happen, we could throw an [InternalError](https://api.rushstack.io/pages/node-core-library.internalerror/) type exception, and then app will instruct the end user to open a GitHub issue. API Extractor uses this approach, because it's trying to be very correct, so we want people to tell us about weird edge cases.

- If it's for development only, you could make a helper function like `logDiagnosticInfo()` that prints in non-red text (`console.log()`), something like `Diagnostic: No resolved entry for dependency`. And then production releases wouldn't print any `Diagnostic:` messages at all. This way it's clear that it's merely informational, it's not an "error"

- Then you could also create a unit test that loads a large example YAML file and asserts that `logDiagnosticInfo()` didn't print anything unexpected.